package NYT.NTableClient.NProto;

import "yt/ytlib/chunk_client/chunk_spec.proto";
import "yt/ytlib/table_client/chunk_meta.proto";

////////////////////////////////////////////////////////////////////////////////

message TDataSliceDescriptor
{
    // Type as in EDataSliceDescriptorType.
    required int32 type = 1;

    // Chunks comprising the input.
    repeated NYT.NChunkClient.NProto.TChunkSpec chunks = 2;

    extensions 100 to max;
}

message TTableSliceDescriptor
{
    extend TDataSliceDescriptor
    {
        optional TTableSliceDescriptor table_slice_descriptor = 100;
    }

    // Table schema.
    required TTableSchemaExt schema = 1;

    // Timestamp (for versioned table).
    optional int64 timestamp = 2 [default = 0];
}

////////////////////////////////////////////////////////////////////////////////

